src/core/dbus-execute.c

```diff
@@ -839,16 +839,8 @@ int bus_exec_context_set_transient_property(
 
                         if (isempty(uu))
                                 c->user = mfree(c->user);
-                        else {
-                                char *t;
-
-                                t = strdup(uu);
-                                if (!t)
-                                        return -ENOMEM;
-
-                                free(c->user);
-                                c->user = t;
-                        }
+                        else if (free_and_strdup(&c->user, uu) < 0)
+                                return -ENOMEM;
 
```

Comment:
actually, this can be reduced even further, as free_and_strdup() also covers the uu == NULL branch internally...